src/connectors/hits/connectHits.ts

```diff
@@ -51,15 +65,26 @@ const connectHits: HitsConnector = function connectHits(
   return widgetParams => {
     const { escapeHTML = true, transformItems = items => items } =
       widgetParams || ({} as typeof widgetParams);
+    let sendEvent;
+    let bindEvent;
 
     return {
       $$type: 'ais.hits',
 
-      init({ instantSearchInstance }) {
+      init({ instantSearchInstance, helper }) {
+        sendEvent = createSendEventForHits({
```

Comment:
The `sendEvent` and `bindEvent` function references are lost between `init` and `render` because they're reassigned. This means that if users bind an event callback in `init` (when `isFirstRender` is true), the callback will be a noop even during `render` (when `isFirstRender` is false).

You can see how we usually fix this by searching for `connectorState` in the codebase.

Reply:
@francoischalifour Can you explain a more how they're reassigned?

Before that, **what** is reassigned? For example, in [`connectBreadcrumb.ts`](https://github.com/algolia/instantsearch.js/blob/feat/send-events/src/connectors/breadcrumb/connectBreadcrumb.ts#L90:L90), `connectorState.createURL` is assigned only once in `init`.

In the current branch, I compared `sendEvent` when isFirstRendering==true and ==false, and they were the same (`===`). Which part am I missing?

Reply:
I think the difference here is that the function isn't also created in `render`, and thus indeed doesn't change reference

Reply:
Had a call with François, and it's is good as is because `sendEvent` is not reassigned in `render` method.